cabal-install/src/Distribution/Client/ProjectOrchestration.hs

```diff
@@ -642,7 +643,7 @@ resolveTargets
       checkTarget :: TargetSelector -> Either (TargetProblem err) [(UnitId, ComponentTarget)]
 
       -- We can ask to build any whole package, project-local or a dependency
-      checkTarget bt@(TargetPackage _ [pkgid] mkfilter)
+      checkTarget bt@(TargetPackage _ (ordNub -> [pkgid]) mkfilter)
```

Comment:
How would it look without viewpatterns?

Reply:
I don't like view patterns in general and especially in this case under a certain level of nesting — they're especially hard to preserve for a human being. I'd suggest to avoid them here. Now, we don't have a policy for it, so you certainly can disregard the suggestion.

Reply:
I really like view patterns ;-)

Reply:
For me, `-XViewPatterns` was like the story of Green Eggs and Ham, by Dr Seuss.